test/coreneuron/unit/mech_mapping/test_mech_mapping.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+# =============================================================================
+# Copyright (c) 2025 Open Brain Institute
+# =============================================================================.
+*/
+#include <catch2/catch_test_macros.hpp>
+#include <iostream>
+
+#include "coreneuron/coreneuron.hpp"
+#include "coreneuron/mechanism/mech_mapping.hpp"
+#include "coreneuron/mechanism/mechanism.hpp"
+#include "coreneuron/permute/data_layout.hpp"
+
+using namespace coreneuron;
+
+// Note: Catch2 does not natively support death tests. I cannot test malformed mechanisms (Katta)
```

Comment:
What is meant by "malformed mechanisms" here? Do you mean mechanisms that don't compile? Because catch2 can catch exceptions, so I assume you mean something else.

Reply:
It is actlually malformed reports that try to find a mechanism. Anything that calls `nrn_abort`. For example, I could as for a report on the mechanism "bau_bau". The name is wrong, the report section is malformed. The problem is that nrn_abort kills everything immediately and I cannot report the error (I think)